Improve RF2 export request performance

nagyo opened the pull request:
- precompute all necessary language codes for each branch path
- precompute all version branches and revision ranges
- use UTC timezone, ISO 8601 format and add current hour in the
effective date of the release archive

nagyo committed:
[rf2] Improve RF2 export request performance

- precompute all necessary language codes for each branch path
- precompute all version branches and revision ranges
- use UTC timezone, ISO 8601 format and add current hour in the
effective date of the release archive

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java:
```diff
@@ -325,8 +338,63 @@ public Rf2ExportResult execute(final RepositoryContext context) {
 	}
 
-	private List<String> computeBranchesToExport(final TreeSet<CodeSystemVersionEntry> versionsToExport) {
+	private Multimap<String, String> getLanguageCodes(RepositoryContext context, Set<String> branchesToExport) {
```
Will this method find language codes on branch sections which are not hit by the exporter later? This may or may not be a problem. 😄

nagyo committed:
[rf2] Do not use sorted set copy during the collection of branches ...

... to be exported

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java:
```diff
@@ -346,80 +414,107 @@ public Rf2ExportResult execute(final RepositoryContext context) {
 		}
 		
-		return EffectiveTimes.format(Dates.todayGmt(), DateFormats.SHORT);
+		return adjustCurrentHour(Dates.parse(Dates.format(new Date(), TimeZone.getTimeZone("UTC"), DateFormats.DEFAULT)));
```
The method name is technically incorrect; not all timezones have full hour differences. 
http://yourcalendricalfallacyis.com/#timezones-always-are-on-the-hour-mark 🤓

nagyo committed:
[rf2] Maintain only a single collection of branches to be exported

- use revision range target if anything needs to be queried before
export
- evaluate existence of 'en' language code as well for each branch /
range

cmark reviewed (approved):
LGTM!
